drivers/flash/flash_stm32_qspi.c

```diff
@@ -311,6 +311,11 @@ static int flash_stm32_qspi_read(const struct device *dev, off_t addr,
 		return -EINVAL;
 	}
 
+	/* read non-zero size */
+	if (!size) {
```

Comment:
nit: better use essentially boolean expressions:

```suggestion
	if (size == 0U) {
```